packages/cdk8s-cli/templates/java-app/.hooks.sscaff.js

```diff
@@ -0,0 +1,32 @@
+const { execSync } = require('child_process');
+const { readFileSync } = require('fs');
+
+const cli = require.resolve('../../bin/cdk8s');
+
+exports.pre = (variables) => {
+  try {
+    execSync('which mvn');
+  } catch {
+    console.error(`Unable to find "mvn". Install from https://maven.apache.org/install.html`);
+    process.exit(1);
+  }
+  variables.constructs_version = variables.constructs_version.replace('^', '').replace('~', '');
+};
+
+exports.post = options => {
+  const { mvn_cdk8s, cdk8s_version } = options;
+  if (!mvn_cdk8s) {
+    throw new Error(`missing context "mvn_cdk8s"`);
+  }
+
+  if (mvn_cdk8s.endsWith('.jar')) {
```

Comment:
add a comment explaining why we need this special case

Reply:
Done!